src/event-replay/helpers.ts

```diff
@@ -0,0 +1,51 @@
+import * as fsr from 'fs-reverse';
+import { waiter } from '../helpers';
+import { PgDataStore } from '../datastore/postgres-store';
+
+/**
+ * Traverse a TSV file in reverse to find the last received `/new_block` node message and return
+ * the `block_height` reported by that event. Even though the block produced by that event might
+ * end up being re-org'd, it gives us a reasonable idea as to what the Stacks node thought
+ * the block height was the moment it was sent.
+ * @param filePath - TSV path
+ * @returns `number` found block height, 0 if not found
+ */
+export async function findTsvBlockHeight(filePath: string): Promise<number> {
+  const blockHeightWaiter = waiter<number>();
+  const reverseStream = fsr(filePath, { flags: 'r' });
+  reverseStream.on('data', data => {
```

Comment:
I don't think this callback approach supports stream backpressure (as in it has unbounded memory usage). 

If `reverseStream` implements the right nodejs stream interface, you could use an async iterator like `for await (const data of reverseStream) { ... }` which should support backpressure and disposal correctly, otherwise pipeline with a transform stream would also work. 

Check out https://nodejs.org/en/docs/guides/backpressuring-in-streams/